spec/controllers/reset_password_controller_spec.rb

```diff
@@ -11,38 +11,36 @@
   end
 
   describe 'POST reset_password#create' do
-    before { create(:user, email: 'test@greenlight.com') }
-
-    it 'generates a unique token and saves its digest for valid emails' do
-      token = 'ZekpWTPGFsuaP1WngE6LVCc69Zs7YSKoOJFLkfKu'
-      allow_any_instance_of(User).to receive(:generate_reset_token!).and_return(token)
+    before do
+      create(:user, email: 'test@greenlight.com')
+      allow_any_instance_of(User).to receive(:generate_reset_token!).and_return('TOKEN')
+      clear_enqueued_jobs
+    end
 
+    it 'generates a unique token, emails and saves its digest for valid emails' do
       post :create, params: { user: { email: 'test@greenlight.com' } }
-      # TODO: Test email delivery.
+      expect(ActionMailer::MailDeliveryJob).to have_been_enqueued.at(:no_wait).exactly(:once).with('UserMailer', 'reset_password_email',
+                                                                                                   'deliver_now', Hash)
```

Comment:
This is a draft until I try to find a simple way to assert the params to ensure that the token/reset url will be sent.

Reply:
If you think this is enough, please tell me